### ISSUE TITLE
VideoJS cannot play low bitrate (144p) outputs from ZenCoder only on Chrome

### ISSUE THREAD

**Issue by brain64bit** [videojs/videojs-contrib-hls#644]
### TL;DR;

Hi i am using two different transcoder for generating 144p segmented HLS using Amazon Elastic Transcoder & Zencoder.  But unfortunately when i play using videojs + videojs-contrib-hls with hls generated by zencoder & elastic transcoder i have different experience:
- 144p from **zencoder** always thrown an error thrown an error only at Chrome Browser, but play nicely at other browser supported MSE ( Firefox > 42), Safari.
```
VIDEOJS: ERROR: (CODE:3 MEDIA_ERR_DECODE) The media playback was aborted due to a corruption problem or because the media used features your browser did not support. 
```
- 144p from **elastic transcoder** play smoothly without any issue on every supported MSE browsers.

---

Our current configurations are:

**Zencoder:**
- video_codec_level: 3
- video_bitrate: 80
- audio_bitrate: 32
- width: 254
- height: 144
- thumbnail_label: thumbnail
- thumbnail_width: 160
- thumbnail_height: 90
- decoder_bitrate_cap: 120
- decoder_buffer_size: 320

**Elastic Transcoder:**
- Codec: H.264
- Codec Options: BufferSize:800,InterlacedMode:Progressive,MaxReferenceFrames:3,Level:3,MaxBitRate:80,ColorSpaceConversionMode:None,Profile:baseline
- Maximum Number of Frames Between Keyframes: 90
- Fixed Number of Frames Between Keyframes: false
- Bit Rate: auto
- Frame Rate: auto
- Video Max Frame Rate: 30
- Max Width: 254
- Max Height: 144
- Sizing Policy: ShrinkToFit
- Padding Policy: NoPad
- Display Aspect Ratio: auto

**Environment:**
- videojs 5.5.1 without flash
- videojs-contrib-hls 1.3.4
- Ubuntu 14.04.2 LTS
- Chrome Version 45.0.2454.101 (64-bit), Firefox 45.0

Here outputs from ffprobe:
**Zencoder HLS Segmented file**
```
{
    "streams": [
        {
            "index": 0,
            "codec_name": "h264",
            "codec_long_name": "H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10",
            "profile": "Constrained Baseline",
            "codec_type": "video",
            "codec_time_base": "1/48",
            "codec_tag_string": "[27][0][0][0]",
            "codec_tag": "0x001b",
            "width": 254,
            "height": 142,
            "coded_width": 256,
            "coded_height": 144,
            "has_b_frames": 0,
            "sample_aspect_ratio": "1:1",
            "display_aspect_ratio": "127:71",
            "pix_fmt": "yuv420p",
            "level": 30,
            "chroma_location": "left",
            "refs": 1,
            "is_avc": "false",
            "nal_length_size": "0",
            "id": "0x101",
            "r_frame_rate": "24/1",
            "avg_frame_rate": "24/1",
            "time_base": "1/90000",
            "start_pts": 900000,
            "start_time": "10.000000",
            "duration_ts": 900000,
            "duration": "10.000000",
            "bits_per_raw_sample": "8",
            "disposition": {
                "default": 0,
                "dub": 0,
                "original": 0,
                "comment": 0,
                "lyrics": 0,
                "karaoke": 0,
                "forced": 0,
                "hearing_impaired": 0,
                "visual_impaired": 0,
                "clean_effects": 0,
                "attached_pic": 0
            }
        },
        {
            "index": 1,
            "codec_name": "aac",
            "codec_long_name": "AAC (Advanced Audio Coding)",
            "profile": "HE-AAC",
            "codec_type": "audio",
            "codec_time_base": "1/48000",
            "codec_tag_string": "[15][0][0][0]",
            "codec_tag": "0x000f",
            "sample_fmt": "fltp",
            "sample_rate": "48000",
            "channels": 2,
            "channel_layout": "stereo",
            "bits_per_sample": 0,
            "id": "0x102",
            "r_frame_rate": "0/0",
            "avg_frame_rate": "0/0",
            "time_base": "1/90000",
            "start_pts": 891240,
            "start_time": "9.902667",
            "duration_ts": 833280,
            "duration": "9.258667",
            "bit_rate": "33937",
            "disposition": {
                "default": 0,
                "dub": 0,
                "original": 0,
                "comment": 0,
                "lyrics": 0,
                "karaoke": 0,
                "forced": 0,
                "hearing_impaired": 0,
                "visual_impaired": 0,
                "clean_effects": 0,
                "attached_pic": 0
            },
            "tags": {
                "language": "und"
            }
        }
    ],
    "chapters": [

    ],
    "format": {
        "filename": "vmb-60s-360p-a-74cb-b144.mp4-00001.ts",
        "nb_streams": 2,
        "nb_programs": 1,
        "format_name": "mpegts",
        "format_long_name": "MPEG-TS (MPEG-2 Transport Stream)",
        "start_time": "9.902667",
        "duration": "10.097333",
        "size": "232368",
        "bit_rate": "184102",
        "probe_score": 100
    }
}
```
**Elastic Transcoder HLS segmented file**
```
{
    "streams": [
        {
            "index": 0,
            "codec_name": "aac",
            "codec_long_name": "AAC (Advanced Audio Coding)",
            "profile": "LC",
            "codec_type": "audio",
            "codec_time_base": "1/44100",
            "codec_tag_string": "[15][0][0][0]",
            "codec_tag": "0x000f",
            "sample_fmt": "fltp",
            "sample_rate": "44100",
            "channels": 2,
            "channel_layout": "stereo",
            "bits_per_sample": 0,
            "id": "0x100",
            "r_frame_rate": "0/0",
            "avg_frame_rate": "0/0",
            "time_base": "1/90000",
            "start_pts": 126000,
            "start_time": "1.400000",
            "duration_ts": 1007281,
            "duration": "11.192011",
            "bit_rate": "30663",
            "disposition": {
                "default": 0,
                "dub": 0,
                "original": 0,
                "comment": 0,
                "lyrics": 0,
                "karaoke": 0,
                "forced": 0,
                "hearing_impaired": 0,
                "visual_impaired": 0,
                "clean_effects": 0,
                "attached_pic": 0
            },
            "tags": {
                "language": "und"
            }
        },
        {
            "index": 1,
            "codec_name": "h264",
            "codec_long_name": "H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10",
            "profile": "Constrained Baseline",
            "codec_type": "video",
            "codec_time_base": "1/48",
            "codec_tag_string": "[27][0][0][0]",
            "codec_tag": "0x001b",
            "width": 254,
            "height": 142,
            "coded_width": 256,
            "coded_height": 144,
            "has_b_frames": 0,
            "sample_aspect_ratio": "6603:6604",
            "display_aspect_ratio": "93:52",
            "pix_fmt": "yuv420p",
            "level": 30,
            "chroma_location": "left",
            "refs": 3,
            "is_avc": "false",
            "nal_length_size": "0",
            "id": "0x101",
            "r_frame_rate": "24/1",
            "avg_frame_rate": "24/1",
            "time_base": "1/90000",
            "start_pts": 130180,
            "start_time": "1.446444",
            "duration_ts": 1012500,
            "duration": "11.250000",
            "bits_per_raw_sample": "8",
            "disposition": {
                "default": 0,
                "dub": 0,
                "original": 0,
                "comment": 0,
                "lyrics": 0,
                "karaoke": 0,
                "forced": 0,
                "hearing_impaired": 0,
                "visual_impaired": 0,
                "clean_effects": 0,
                "attached_pic": 0
            }
        }
    ],
    "chapters": [

    ],
    "format": {
        "filename": "vmb-60s-360p-a-f2c7-b144.mp400000.ts",
        "nb_streams": 2,
        "nb_programs": 1,
        "format_name": "mpegts",
        "format_long_name": "MPEG-TS (MPEG-2 Transport Stream)",
        "start_time": "1.400000",
        "duration": "11.296444",
        "size": "241768",
        "bit_rate": "171217",
        "probe_score": 100
    }
}
```

Please help, Thanks.

**Comment by t-n** [comment]
There is a bug in Chromium regarding playing HE-AAC streams with an AAC-LC header, which is vaild according to the standard but breaks in Chrome. (https://bugs.chromium.org/p/chromium/issues/detail?id=534301)
You can easily test to see if this is the issue by editing the m3u8 and changing all mp4a.40.2 to mp4a.40.5. 
If it works you either have to update all m3u8 or re-encode with the correct settings.
Or you could do something like this, but only for the streams where it's needed,
```
    let ParseStream = videojs.m3u8.ParseStream;
    ParseStream.prototype._originalPush = ParseStream.prototype.push;
    ParseStream.prototype.push = function(line) {
      line = line.replace(/mp4a\.40\.2/, 'mp4a.40.5');
      this._originalPush(line);
    };
```
And it actually seems that this bug is finally fixed in Chrome Canary.

**Comment by dmlap** [comment]
I think you nailed it, @t-n. For now-- make sure your streams have proper codec declarations. Once the Chrome fix hits general release, this won't be such an issue anymore.

**Comment by brain64bit** [comment]
For zencoder if you want to provide 144p (audio bitrate < 80 kbps) rendition by default it will be use HE-AAC, fortunately there is option for forcing aac profile to be use **AAC-LC** just pass an option:

    max_aac_profile: "aac-lc"

Thanks
